### PR TITLE
Balanced brackets

### PR DESCRIPTION
**Pull request by privat**
The PR started as a simple implementation for the task http://rosettacode.org/wiki/Balanced_brackets but some improvements on the lib and tests.sh where also added

### CODE REVIEW

File: examples/rosettacode/balanced_brackets.nit [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Task: Balanced brackets
+# SEE: <http://rosettacode.org/wiki/Balanced_brackets>
```
Comment (Morriar) [inline examples/rosettacode/balanced_brackets.nit]: URLs are automatically parsed even without the `<>`.

File: lib/standard/environ.nit [hunk 2]
```diff
@@ -60,3 +60,11 @@ redef class NativeString
 	private fun get_environ: NativeString is extern "string_NativeString_NativeString_get_environ_0"
 	private fun setenv( v : NativeString ) is extern "string_NativeString_NativeString_setenv_1"
 end
+
+redef class Sys
+	redef init
+	do
+		var x = "NIT_SRAND".environ
+		if x != "" then srand_from(x.to_i)
```
Comment (Morriar) [inline lib/standard/environ.nit]: Use `is_empty` instead?

File: lib/standard/math.nit [hunk 3]
```diff
@@ -252,6 +257,19 @@ redef class Collection[ E ]
...
+
+	# Return a new array made of elements in a random order.
+	#
+	# ~~~
+	# var a = [1,2,1].to_shuffle
+	# assert a == [1,1,2] or a == [1,2,1] or a == [2,1,1]
+	# ~~~
+	fun to_shuffle: Array[E]
+	do
+		var res = self.to_a
```
Comment (Morriar) [inline lib/standard/math.nit]: useless `self`?

**Comment by Morriar** [comment]
+1

**Comment by R4PaSs** [comment]
:shipit:

**Comment by privat** [comment]
reroll: try to pass tests

**Pull request merged**